Xamarin.Forms.Platform.UAP/ListViewRenderer.cs

```diff
@@ -102,7 +102,11 @@ void ReloadData()
 			}
 			else
 			{
-				_collectionIsWrapped = !IsObservableCollection(Element.ItemsSource);
+				// if the collection is not implemented `IList`, then 
+				// each change will reset the entire collection (`Clear` action ListProxy class).
+				var isCorrectCustomCollection = Element.ItemsSource is IList && Element.ItemsSource is INotifyCollectionChanged;
+
+				_collectionIsWrapped = !(isCorrectCustomCollection || IsObservableCollection());
```

Comment:
Since `ObservableCollection<T>` implements `INotifyCollectionChanged` and `IList` (via `Collection<T>`), I don't think this needs the `IsObservableCollection` check.